File: 2025-and-before/0290-word-pattern/0290-word-pattern.cpp
class Solution {
public:
    bool wordPattern(string pattern, string s) {
        
        
        s+=" ";
        unordered_map<string,char>mp;
        unordered_set<char>st;
        int c = 0;
        string temp = "", currPattern ="";
        
        for(char&ch : s){
            
            if(ch == ' '){
                if(mp.count(temp) == 0)
                {
                    if(st.count(pattern[c]) == 0)
                    {
                        mp[temp] = pattern[c];
                        st.insert(pattern[c]);
                    }
                    
                    else 
                        return false;
                }
                
                currPattern+=mp[temp];
                temp="";
                c++;
            }
            
            else{
                temp+=ch;
            }
        }
        
        //for(auto&it : mp){
        //      cout<<it.first<<" "<<it.second<<endl;
        // }
        // cout<<currPattern<<endl;
        return currPattern == pattern;
    }
};

File: 2025-and-before/0290-word-pattern/0290-word-pattern.py
class Solution:
    def wordPattern(self, pattern: str, s: str) -> bool:
        
        s = s.split();
        
        if len(pattern) != len(s):
            return False;
        
        i=0;
        mp={};
        st=set();
        currPattern="";
        
        for str in s:
            if(str not in mp):
                if(pattern[i] not in st):
                    mp[str] = pattern[i];
                    st.add(pattern[i]);
                else :
                    return False;
            
            currPattern = currPattern + mp[str];
            i=i+1;
            
        
        if(currPattern == pattern):
            return True;
        else:
            return False;
        